micro-rdk/src/common/ota.rs

```diff
@@ -525,18 +525,25 @@ impl<S: OtaMetadataStorage> OtaService<S> {
                             #[cfg(feature = "esp32")]
                             {
                                 log::info!("verifying new ota firmware");
-                                let mut loader = EspFirmwareInfoLoader::new();
-                                loader
-                                    .load(&data)
+                                let mut new_fw = FirmwareInfo {
```

Comment:
note: can't use `..Default::default()` because `FirmwareInfo` does not  `impl Default`